src/plugins/analysis/ipc/code/ipc_analyzer.py

```diff
@@ -3,44 +3,69 @@
 import json
 import tempfile
 from pathlib import Path
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, List, Union
 
 from docker.types import Mount
+from pydantic import BaseModel, Field
+from semver import Version
 
-from analysis.PluginBase import AnalysisBasePlugin
+from analysis.plugin import AnalysisPluginV0
+from analysis.plugin.compat import AnalysisBasePluginAdapterMixin
 from helperFunctions.docker import run_docker_container
 
 if TYPE_CHECKING:
-    from objects.file import FileObject
+    from io import FileIO
 
 DOCKER_IMAGE = 'ipc'
 
 
-class AnalysisPlugin(AnalysisBasePlugin):
-    """
-    Inter-Process Communication Analysis
-    """
+class IpcTarget(BaseModel):
+    name: str = Field(description='target of the IPC (usually a file path or address)')
+    ipc_calls: List[IpcCall]
 
-    NAME = 'ipc_analyzer'
-    DESCRIPTION = 'Inter-Process Communication Analysis'
-    VERSION = '0.1.1'
-    FILE = __file__
 
-    MIME_WHITELIST = [  # noqa: RUF012
-        'application/x-executable',
-        'application/x-object',
-        'application/x-sharedlib',
-    ]
-    DEPENDENCIES = ['file_type']  # noqa: RUF012
-    TIMEOUT = 600  # 10 minutes
+class IpcCall(BaseModel):
+    type: str = Field(description='a type of call (usually a POSIX system call)')
+    arguments: List[Union[str, int, List[Union[str, int]]]] = Field(description='arguments of the IPC call')
 
-    def _run_ipc_analyzer_in_docker(self, file_object: FileObject) -> dict:
+
+class AnalysisPlugin(AnalysisPluginV0, AnalysisBasePluginAdapterMixin):
+    class Schema(BaseModel):
+        targets: List[IpcTarget] = Field(description='An array of targets of IPC calls')
```

Comment:
Wouldn't it be useful to make this a dictionary?
E.g. to find out which IPC calls target a particular pipe one could simply check whether the path is in the targets dictionary.

Reply:
I don't think so: JSON objects are not intended to be used with unknown keys -- that's what arrays are for.
> E.g. to find out which IPC calls target a particular pipe one could simply check whether the path is in the targets dictionary.
If you want to find them using a Postgres or GraphQL query, it doesn't really matter, because it works either way.